framework/src/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -221,19 +225,96 @@ public int hashCode() {
         return result;
     }
 
-    /** Returns the annotation mirror as specified by the "qualifier" value in metaAnno. */
-    private AnnotationMirror getAnnotationMirrorOfQualifier(AnnotationMirror metaAnno) {
+    /**
+     * Returns the annotation mirror as specified by the "qualifier" value in {@code qualifierAnno}.
```

Comment:
Please clarify what is the "qualifier" value.  Is that the value of a field named `qualifier`, or something else?